qobuz_importer.user.js

```diff
@@ -26,19 +26,50 @@ if (DEBUG) {
 }
 
 // list of qobuz artist id which should be mapped to Various Artists
-var various_artists_ids = [26887, 145383, 353325, 183869, 997899];
+var various_artists_ids = [26887, 145383, 353325, 183869, 997899, 2225160],
+    various_composers_ids = [573076];
+
+function isVariousArtists(artist) {
+    // Check hard-coded various artist ids
+    if ($.inArray(artist.id, various_artists_ids) != -1 || $.inArray(artist.id, various_composers_ids) != -1) {
+        return true;
+    } else if ($.inArray(artist.slug, ['various-artist', 'various-composers']) != -1) {
+        // Let's assume various based on the slug
+        return true;
+    }
+    return false;
+}
+
+function getPerformers(trackobj) {
+    return (
+        (typeof trackobj.performers !== 'undefined' &&
+            trackobj.performers
+                .replace('\r', '')
+                .split(' - ')
+                .map(function(v) {
+                    let list = v.split(', ');
+                    let name = list.shift();
+                    return [name, list];
+                })) || [[trackobj.performer.name, ['Primary']]]
+    );
+}
 
 function parseRelease(data) {
-    let release = {};
+    let release = {},
+        classical = false;
 
     release.script = 'Latn';
     release.url = `https://www.qobuz.com${data.relative_url}`; // no lang
 
     release.title = data.title;
-    release.artist_credit = MBImport.makeArtistCredits([data.artist.name]); // FIXME: various artists
-    if ($.inArray(data.artist.id, various_artists_ids) != -1) {
-        release.artist_credit = [MBImport.specialArtist('various_artists')];
-    }
+    if (isVariousArtists(data.artist)) {
```

Comment:
So it only works for Various Artists albums having "Classique" as genre ?
What about non-french Qobuz websites ? And non-various artists classical albums ?

https://www.qobuz.com/de-de/album/brahms-piano-quintet-piano-quartets-nos-1-3-quatuor-hermes-and-geoffroy-couteau/rwp9d27ev3ytb

I wonder if having a checkbox allowing the user to decide wouldn't be better (and the checkbox can be checked by auto detectiion).

Reply:
Yes, it's using composer for artist just in classical various releases now. It would probably be better to detect the classical before various, and use composer when artist is missing. As with the link you gave, there is no artist, only performer and composer, so its' artist is the ensemble.

Classique seems to be constant in the genres_list from the API, so it is working at least in gb-en, fr-fr and de-de. There is store_related object with translations. And trying some of those store locales, some work, some require to be registered to work, some give 404 (seem that there are some georestrictions both to releases and stores). 

The checkbox with auto detection idea is great. At the moment the parsing to the mb release is done on load, and to be able to parse it again, the data object needs to be stored.

I think I'll implement both of those.